### DISCUSSION TITLE
Pre token generation Lambda trigger V2_0 events

### DISCUSSION THREAD

**Discussion by ernest-folch-fleksy** [aws/aws-lambda-dotnet#1650] [Q&A] [answered]
Checking the classes inside Amazon.Lambda.CognitoEvents library, it seems it is only supported the V1_0 events. Is there any way to deal with V2_0 events?

**Comment by ashishdhingra** [comment, marked as answer]
@ernest-folch-fleksy Good morning. Do you have reference links for Cogito v2.0 events? 
- If these are new event types, then these could be added to existing https://github.com/aws/aws-lambda-dotnet/tree/master/Libraries/src/Amazon.Lambda.CognitoEvents package.
- If these have separate shape for existing events, perhaps we would need to create a new package (on the similar lines as https://github.com/aws/aws-lambda-dotnet/tree/master/Libraries/src/Amazon.Lambda.LexV2Events).

Feel free to submit a new feature request in this repository with all the necessary details which could be reviewed by the team.

Thanks,
Ashish

**Reply by ernest-folch-fleksy** [reply]
Hi, as you suggested I've created the pull request: https://github.com/aws/aws-lambda-dotnet/pull/1656.

Regards,
Ernest
